distributed/comm/ucx.py

```diff
@@ -222,7 +223,8 @@ def peer_address(self) -> str:
 
                 # It is necessary to first populate `frames` with CUDA arrays and synchronize
                 # the default stream before starting receiving to ensure buffers have been allocated
-                synchronize_stream(0)
+                if any(is_cudas.tolist()):
```

Comment:
Can just use this reduction with the NumPy array.

```suggestion
                if is_cudas.any():
```